src/Model/CompositeTypeGo.cs

```diff
@@ -261,6 +261,15 @@ public virtual string Fields()
                 {
                     indented.Append($"{property.Name} - {property.Documentation}".ToCommentBlock());
                 }
+                if (property.Deprecated)
+                {
+                    var message = "This property has been deprecated.";

```

Comment:
this is the kind of code I wanted to prevent 😉 

From the [issue](https://github.com/Azure/autorest/issues/2812):
> You are guaranteed that `Deprecated == true` exactly if `DeprecationMessage != null`, i.e. there is a deprecation message exactly if an entity is deprecated - *use that message* and do not come up with your own messages! If the messages I give your generator suck, I'd be happy to adjust them.

So this block is equivalent to 
``` C#
if (property.Deprecated) {
    indented.Append($"//\n// Deprecated: {property.DeprecationMessage}\n");
}
```

Reply:
same pattern appears more often in this PR... preventing every single generator to do this is exactly what I wanted to prevent (imagine the bug potential, forgetting to put in a default message in 500 places) 🙂

Reply:
@olydis if you look we use taht only if the Deprecation Message is null. If that is never null this will never happen.

Reply:
that's exactly my point

Reply:
@olydis also for some types, they are not coming from swagger models, for future types we generate them for methods. So we wanted that if an operation is deprecated the corresponding Future to be marked as deprecated.

Reply:
saw that, no problem with that 😋